jwst/resample/resample_spec.py

```diff
@@ -142,20 +142,20 @@ def build_interpolated_output_wcs(self, refmodel=None):
             # first input model sets intializes wavelength array and defines
             # the spatial scale of the output wcs
             if im == 0:
-                for iw in wavelength_array:
-                    all_wavelength.append(iw)
+                all_wavelength = np.append(all_wavelength, wavelength_array)
 
+                # find the center ra and dec for this slit at central wavelength
                 lam_center_index = int((bb[spectral_axis][1] -
                                         bb[spectral_axis][0]) / 2)
-                if spatial_axis == 0:
-                    ra_center = ra[lam_center_index, :]
-                    dec_center = dec[lam_center_index, :]
+                if spatial_axis == 0:  # MIRI LRS, the WCS x axis is spatial
+                    ra_slice = ra[lam_center_index, :]
+                    dec_slice = dec[lam_center_index, :]
                 else:
-                    ra_center = ra[:, lam_center_index]
-                    dec_center = dec[:, lam_center_index]
-                # find the ra and dec for this slit using center of slit
-                ra_center_pt = np.nanmean(ra_center)
-                dec_center_pt = np.nanmean(dec_center)
+                    ra_slice = ra[:, lam_center_index]
+                    dec_slice = dec[:, lam_center_index]
+                # wrap RA if near zero
+                ra_center_pt = np.nanmean(wrap_ra(ra_slice))
```

Comment:
This was the bug.  For RA~0, the mean of the RA slice was the mean of zeros and 359, so an output frame centered at RA~180 deg.